tests/tst_ncio.F90

```diff
@@ -235,6 +235,13 @@ program tst_ncio
   endif
 
   print *,'*** Test reading of data just written...'
+
+  call read_vardata(dset, 'grid_xt', values_1d)
+  if (maxval(values_1d) .ne. 358.59375) then
```

Comment:
Checking the max value is not a great test. Many things can be wrong - like every other value could be zero.

Instead, load an array of test data as constants in the program and test the first 100 values...